### PR TITLE
move pio parameters to nuopc.runconfig input file

### PR DESCRIPTION
**Pull request by jedwards4b**
### Description of changes
Update shr_pio_mod, removing mct centric initialization.
Improve logging and make sure logs go to the correct component.
Depends on cime branch async_io_in_esmf

### Specific notes

Contributors other than yourself, if any:

CMEPS Issues Fixed (include github issue #):

Are changes expected to change answers? bfb

Any User Interface Changes (namelist or namelist defaults changes)?
The modelio namelist has been eliminated and the parameters from that file were moved to nuopc.runconfig,
The xml variable PIO_ASYNC_INTERFACE was changed from a global variable to one that could be set individually
for each component. 

### Testing performed

Testing performed if application target is CESM:
- [X] (recommended) CIME_DRIVER=nuopc scripts_regression_tests.py
   - machines: cheyenne intel 
   - details (e.g. failed tests): IRT_N2_Vmct_Ln9.f19_g16_rx1.A.cheyenne_intel fails, apparently expected.

- [ ] (recommended) CESM testlist_drv.xml
   - machines and compilers:
   - details (e.g. failed tests):
- [x] (optional) CESM prealpha test
   - machines and compilers
   - details (e.g. failed tests):
- [ ] (other) please described in detail
   - machines and compilers
   - details (e.g. failed tests):

Testing performed if application target is UFS-coupled:
- [ ] (recommended) UFS-coupled testing
   - description:
   - details (e.g. failed tests):

Testing performed if application target is UFS-HAFS:
- [ ] (recommended) UFS-HAFS testing
   - description:
   - details (e.g. failed tests):

### Hashes used for testing:

- [ ] CESM:
  - repository to check out: https://github.com/ESCOMP/CESM.git
  - branch/hash:
- [ ] UFS-coupled, then umbrella repostiory to check out and associated hash:
  - repository to check out:
  - branch/hash:
- [ ] UFS-HAFS, then umbrella repostiory to check out and associated hash:
  - repository to check out:
  - branch/hash:

### CODE REVIEW

File: mediator/med_io_mod.F90 [hunk 1]
```diff
@@ -177,6 +177,7 @@ subroutine med_io_init(gcomp, rc)
     pio_iotype   =  shr_pio_getiotype(med_id)
     pio_ioformat =  shr_pio_getioformat(med_id)
 #else
+    print *,__FILE__,__LINE__,'PIO type, format:',pio_iotype, pio_ioformat
```
Comment (mvertens) [inline mediator/med_io_mod.F90:180]: Please remove this!

**Review by mvertens — CHANGES_REQUESTED** [review]

**Comment by mvertens** [comment]
@jedwards4b - are we ready to merge this?

**Comment by jedwards4b** [comment]
@mvertens There is a failing test I need to fix first, see: https://github.com/ESMCI/cime/pull/4199

**Comment by mvertens** [comment]
@jedwards4b - please summarize the testing you did for this PR. I cannot accept that before then. Also please fill in a summary of the user interface changes and other requested input for the PR.

**Review by mvertens — APPROVED** [review]

**Pull request merged**